messaging/src/main/java/org/axonframework/commandhandling/gateway/NonTransientClassesPredicate.java

```diff
@@ -0,0 +1,49 @@
+package org.axonframework.commandhandling.gateway;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * A predicate for checking non-transiency of a failure comparing it against configured concrete classes.
+ * <p/>
+ * Comparison checks if any of configured classes is assignable from a tested failure instance.
+ * <p/>
+ * This predicate is intended to be used for configuring {@link RetryScheduler} implementations like in the following example:
+ * <pre>
+ * IntervalRetryScheduler myRetryScheduler = IntervalRetryScheduler
+ *      .builder()
+ *      .retryExecutor(new ScheduledThreadPoolExecutor(1))
+ *      .nonTransientFailurePredicate(
+ *          new NonTransientClassesPredicate(
+ *              AxonNonTransientException.class, NullPointerException.class, IllegalArgumentException.class,
+ *              IllegalStateException.class
+ *          )
+ *      )
+ *      .build();
+ * </pre>
+ */
+public class NonTransientClassesPredicate implements Predicate<Throwable> {
+  private final List<Class<? extends Throwable>> nonTransientFailures;
+
+  /**
+   * Initialize the predicate with class(es) that are considered as non-transient.
+   *
+   * @param nonTransientFailures vararg list of non-transient class(es)
+   */
+  @SafeVarargs
+  public NonTransientClassesPredicate(Class<? extends Throwable> ...nonTransientFailures) {
+    this.nonTransientFailures = Arrays.asList(nonTransientFailures);
+  }
+
+  @Override
+  public boolean test(Throwable failure) {
```

Comment:
Nit: why not include a separate method, something like `isNonTransient`, which is invoked from the `test`? 

That way you have a named function that can include some JavaDoc on the exact process performed.
Added, it abstracts away the "Predicate-ness" a little.

Reply:
I added it to the latest commits. Please see if this is what you had in mind.